### PR TITLE
[fix](modify) permit modify default value

### LINKED ISSUE #17045: [Enhancement]  change default value
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

doris does not allow users to change default value of a column.

### Solution

allow change default value of new data.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by linzs148**
# Proposed changes

Issue Number: close #17045

## Problem summary

doris does not allow users to change default value of a column.

## Checklist(Required)

* [√ ] Does it affect the original behavior
* [√ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.03 seconds
 stream load tsv:          473 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         38 seconds loaded 2358488459 Bytes, about 59 MB/s
 stream load orc:          75 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230307172824_clickbench_pr_110251.html

File: regression-test/suites/dynamic_table_p0/test_dytable_alter_default_value.groovy [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        sql "USE ${database_name}"
+        sql "DROP TABLE IF EXISTS ${table_name}"
+        sql """
+            CREATE TABLE IF NOT EXISTS ${table_name} (
+                id BIGINT,
+                `${column_name}` ${src_type} DEFAULT '${src_value}'
+            )
+            DUPLICATE KEY(`id`)
+            DISTRIBUTED BY HASH(`id`) BUCKETS 10
+            properties("replication_num" = "1");
```
Comment (yiguolei) [inline regression-test/suites/dynamic_table_p0/test_dytable_alter_default_value.groovy:29]: should disable compaction. Or the rowset is merged and the result may unstable.

File: regression-test/suites/dynamic_table_p0/test_dytable_alter_default_value.groovy [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        """
+        for (i = 0; i < 10; i++) {
+            sql "INSERT INTO ${table_name} VALUES (${i}, DEFAULT)"
+        }
+        sql "ALTER TABLE ${table_name} MODIFY COLUMN ${column_name} ${dst_type} DEFAULT '${dst_value}'"
+        Thread.sleep(3000)
+        for (i = 10; i < 20; i++) {
```
Comment (yiguolei) [inline regression-test/suites/dynamic_table_p0/test_dytable_alter_default_value.groovy:36]: add test for delete where on column that has default value.

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: regression-test/suites/dynamic_table_p0/test_dytable_alter_default_value.groovy [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        def result2 = sql "SELECT COUNT(*) FROM ${table_name} WHERE ${column_name} = '${src_value}'"
+        assertEquals(result2[0][0], 10)
+        def result3 = sql "SELECT COUNT(*) FROM ${table_name} WHERE ${column_name} = '${dst_value}'"
+        assertEquals(result3[0][0], 10)
+        // step8: delete where column == dst_value
+        sql "DELETE FROM ${table_name} WHERE ${column_name} = '${dst_value}'"
+        def result4 = sql "SELECT COUNT(*) FROM ${table_name} WHERE ${column_name} = '${src_value}'"
+        assertEquals(result4[0][0], 10)
+        def result5 = sql "SELECT COUNT(*) FROM ${table_name} WHERE ${column_name} = '${dst_value}'"
+        assertEquals(result5[0][0], 0)
```
Comment (yiguolei) [inline regression-test/suites/dynamic_table_p0/test_dytable_alter_default_value.groovy:68]: should check the actual data not only the row count, it is no accurate.

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Pull request closed without merging**